keras/backend/mxnet_backend.py

```diff
@@ -2038,8 +2044,12 @@ def concatenate(tensors, axis=-1):
         else:
             axis = 0
 
-    tensors = [t.symbol for t in tensors]
-    return KerasSymbol(mx.sym.concat(*tensors, dim=axis))
+    symbols = [t.symbol for t in tensors]
+
+    if py_all([is_sparse(t) for t in tensors]):
+        return KerasSymbol(mx.sym.sparse.concat(*symbols, dim=0))
+
+    return KerasSymbol(mx.sym.concat(*symbols, dim=axis))
```

Comment:
If part of tensors are sparse, so it will fail the check at `py_all([is_sparse(t) for t in tensors]):`, what will happen if you pass them to dense concat?